### PR TITLE
Pull Vitess image when we know if it's a Vitess data source or not

### PR DESCRIPTION
**Pull request by tirsen**

### CODE REVIEW

File: misk-hibernate/src/main/kotlin/misk/vitess/StartVitessService.kt [hunk 1]
```diff
@@ -313,14 +313,32 @@ class LogContainerResultCallback : ResultCallbackTemplate<LogContainerResultCall
...
+    this.qualifier = qualifier
+    this.environment = environment
+    this.config = config
+    this.consumedKeys = setOf()
+    this.producedKeys = setOf(Key.get(StartVitessService::class.java))
+
+    if (config.type == DataSourceType.VITESS) {
+      // Need to pull the docker image here,
+      // can't do it during service start up because it's too slow
+      StartVitessService.pullImage()
```
Comment (tirsen) [inline misk-hibernate/src/main/kotlin/misk/vitess/StartVitessService.kt]: I have to confess that I don't really like doing things like this in a constructor, but neither do I like doing it in a class initializer either so I guess this doesn't really make things worse either.

Comment (mightyguava) [reply misk-hibernate/src/main/kotlin/misk/vitess/StartVitessService.kt]: Your other option is to do it in the `configure()` function of the vitess module. That might be slightly better?

File: misk-hibernate/src/main/kotlin/misk/vitess/StartVitessService.kt [hunk 2]
```diff
@@ -329,6 +347,10 @@ class StartVitessService(
       // We only start up Vitess if Vitess has been configured
       return
     }
+    check(StartVitessService.imagePulled.get()) {
+      "Docker image not pulled, we can't pull here because service start up times out, " +
```
Comment (mightyguava) [inline misk-hibernate/src/main/kotlin/misk/vitess/StartVitessService.kt]: If it's in the constructor, you shouldn't need this?

**Comment by chris-ryan-square** [comment]
If this PR is still relevant, then please fix conflicts, rebase and get a green CI build.
If the PR is no longer relevant please close.

This PR will be closed at the end of January 2021 otherwise.

**Comment by chris-ryan-square** [comment]
Closing request as per previous announcement.

**Pull request closed without merging**